crates/agent/src/discovers.rs

```diff
@@ -166,38 +168,63 @@ impl DiscoverHandler {
         )
         .await?;
 
-        match result {
-            Ok(catalog) => {
-                draft::upsert_specs(row.draft_id, catalog, txn)
-                    .await
-                    .context("inserting draft specs")?;
-                let detail = format!(
-                    "system created publication in response to discover: {}",
-                    row.id
-                );
-                // If requested, create a publication of this draft.
-                let publication_id = if row.auto_publish {
-                    let id = agent_sql::publications::create(
-                        txn,
-                        row.user_id,
-                        row.draft_id,
-                        row.auto_evolve,
-                        detail,
-                    )
-                    .await?;
-                    Some(id)
-                } else {
-                    None
-                };
-
-                Ok((row.id, JobStatus::Success { publication_id }))
-            }
+        let mut catalog = match result {
+            Ok(cat) => cat,
             Err(errors) => {
                 draft::insert_errors(row.draft_id, errors, txn).await?;
-
-                Ok((row.id, JobStatus::MergeFailed))
+                return Ok((row.id, JobStatus::MergeFailed));
             }
+        };
+
+        // Prune out any unchanged specs, but only if we're automatically publishing.
+        // If a user is interactively discovering, then the UI will need all the specs,
```

Comment:
Does it? I'm missing something. Can't the UI use the existing spec? That's what it appears to do today, when you edit an existing capture and review a bound collection.

I'm thinking a) it would be nice to consolidate these branches, and b) publication history does get cluttered with no-op publications for collections that are touched but unchanged by a manual discover, so might as well prune those too.

Reply:
That's a good point. I guess I just assmumed they'd need to be a part of the draft in order for the UI to work, but if that's not the case, then I agree it'd be cleaner to just always prune them. I'll verify the UI behavior and make it unconditional if I can.